### PR TITLE
fix(tv show tabs): add sorting to vuex store to fix season ordering

### LINKED ISSUE #479: Episodes are in the wrong season
Episodes are part of the wrong seasons. (Using the latest 10.7.0-rc2 server)

Alf in web:
image

Alf in Vue (Should be season 1):
image

Archer in web:
image

Archer in Vue (Should be season 3):
image

### PR DESCRIPTION
**Pull request by Camc314**
Potentially fixes #479

After adding the array of episodes of each season. Sort by parentIndexNumber

### CODE REVIEW

**Review by ferferga — APPROVED** [review]

**Comment by MrTimscampi** [comment]
Analyzing this a bit, this seems more like a logic issue in how the data is structured and assigned than a sorting issue.

We're assigning episodes to an array, by combining the existing contents of the array with the new one. However, *nothing* guarantees that we are going to receive the seasons in the proper order or that said seasons would contain anything, or even return at all.

I think the proper fix for this would be to rework the structure to be like so:
```typescript
const TvShowsState = {
  '494947c860b2d2abaa645bffb9fde158': {
    'seasons': {
      0: {
        'information': BaseItemDto // The season's info
        'episodes': BaseItemDto[] // An array of all the episodes
      }
    }
  }
}
```

This way, we can set the info by key instead of by index, which allows us to go from 0 to 4 back to 2, as items come in. It also sets us up for adding more info to this store at a later date (Like storing the series' info directly in the store).

**Review by MrTimscampi — CHANGES_REQUESTED** [review]
Should've probably marked my previous comment as a request for change, as the current way of doing things won't solve the core issue we're having.

**Comment by Camc314** [comment]
Finally updated to use an object. Should be working now

```
state: {
  'seriesId': {
    seasons: []
    seasonEpisodes: {
      'season-id': [],
      'season-id': []
    }
  }
}

File: store/tvShows.ts [hunk 1]
```diff
@@ -5,42 +5,42 @@ import {
...
 type MutationPayload = {
   seasons: BaseItemDto[];
   seasonEpisodes: BaseItemDto[];
   itemId?: string;
+  seasonItemId?: string;
 };
 
-export const getters: GetterTree<TvShowsState, AppState> = {
+export const getters: GetterTree<TvShowsState, TvShowsState> = {
   getSeasons: (state) => ({ itemId }: { itemId: string }): BaseItemDto[] => {
```
Comment (ThibaultNocchi) [inline store/tvShows.ts:40]: Why is it like this and not like this:
```suggestion
  getSeasons: (state) => ( itemId: string ): BaseItemDto[] => {
```

File: store/tvShows.ts [hunk 2]
```diff
@@ -5,42 +5,42 @@ import {
...
     return state[itemId]?.seasons;
   },
   getSeasonEpisodes: (state) => ({
     itemId
   }: {
     itemId: string;
-  }): BaseItemDto[][] => {
```
Comment (ThibaultNocchi) [inline store/tvShows.ts:43]: Same here

**Review by ThibaultNocchi — COMMENTED** [review]
You can also simplify your Vue.set calls by just editing a sub-prop such as there: https://github.com/jellyfin/jellyfin-vue/blob/e70915b5621541cf957c2109d43cd03c62ba857b/store/displayPreferences.ts#L96

**Comment by Camc314** [comment]
> You can also simplify your Vue.set calls by just editing a sub-prop such as there:
> 
> https://github.com/jellyfin/jellyfin-vue/blob/e70915b5621541cf957c2109d43cd03c62ba857b/store/displayPreferences.ts#L96

The only issue is that `state[itemId]` may not be defined yet. By setting the whole state, we create the property on `state` if it is not already defined

**Comment by ThibaultNocchi** [comment]
Yeah that seems fair then

**Comment by ThibaultNocchi** [comment]
@Camc314 Also did you see the small comments I left? :)

**Comment by Camc314** [comment]
> @Camc314 Also did you see the small comments I left? :)


Yep, I'll address them in the next few hours 👍

File: store/tvShows.ts (hunk 1 above)
Comment (Camc314) [reply store/tvShows.ts:40]: My preference is to pass objects and then destructure them, it makes it easier if we have multiple items and only certain ones need to be defined, but I guess we don't need that here.

Comment (ThibaultNocchi) [reply store/tvShows.ts]: @Camc314 That's fair in those cases, but yeah there I don't see the need :p thanks!

**Review by ThibaultNocchi — APPROVED** [review]

**Review by ferferga — APPROVED** [review]

**Pull request merged**